### PR TITLE
Readability revisions for #109891

### LINKED ISSUE #110079: [macOS] Bring up "hello_world" devicelab, compilation test for x86.

### PR DESCRIPTION
**Pull request by a-wallen**
#### Related
fixes https://github.com/flutter/flutter/issues/110079

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 1]
```diff
@@ -1398,18 +1398,19 @@ class CompileTest {
...
-        if (appBundlePath == null) {
+        if (appPath == null) {
           throw 'Failed to find app bundle in ${buildDirectory.path}';
         }
-        // IPAs are created manually, https://flutter.dev/ios-release/
-        await exec('tar', <String>['-zcf', 'build/app.ipa', appBundlePath]);
+        // Validate changes in Dart snapshot format and data layout do not
+        // change compression size.
+        await exec('tar', <String>['-zcf', 'build/app.ipa', appPath]);
         releaseSizeInBytes = await file('$cwd/build/app.ipa').length();
```
Comment (jmagman) [inline dev/devicelab/lib/tasks/perf_tests.dart:1409]: Remove `.ipa extension`
```
        await exec('tar', <String>['-zcf', 'build/app.tar.gz', appPath]);
        releaseSizeInBytes = await file('$cwd/build/app.tar.gz').length();
```

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 2]
```diff
@@ -1398,18 +1398,19 @@ class CompileTest {
...
+        final String? appPath =
             _findDarwinAppInBuildDirectory(buildDirectory.path);
-        if (appBundlePath == null) {
+        if (appPath == null) {
           throw 'Failed to find app bundle in ${buildDirectory.path}';
         }
-        // IPAs are created manually, https://flutter.dev/ios-release/
-        await exec('tar', <String>['-zcf', 'build/app.ipa', appBundlePath]);
+        // Validate changes in Dart snapshot format and data layout do not
+        // change compression size.
```
Comment (jmagman) [inline dev/devicelab/lib/tasks/perf_tests.dart:1407]:
```suggestion
        // change compression size. This also simulates the size of an IPA on iOS.
```

File: dev/devicelab/lib/tasks/perf_tests.dart (hunk 1 above)
Comment (a-wallen) [reply dev/devicelab/lib/tasks/perf_tests.dart]: Good catch!

**Review by loic-sharma — APPROVED** [review]

**Review by jmagman — APPROVED** [review]
(I shouldn't have been assigned, just kept as a reviewer).
LGTM

**Pull request merged**